.cirrus.yml

```diff
@@ -3,11 +3,13 @@
 # SPDX-License-Identifier: MIT
 
 test_template: &test
+  environment:
+    CODECOV_TOKEN: ENCRYPTED[e18180139fa0a70726bb6f8dfbbc9fd53da36e6a4d2b449e30f99a255fdfdab63863117e8b6665deb06dd99d95e5aaad]
   test_script:
     - command -v python3
     - python3 --version
-    - python3 -m pip install .[test]
-    - python3 -m pytest --showlocals -vv
+    - python3 -m pip install .[test] pytest-codecov
+    - python3 -m pytest --showlocals -vv --cov --cov-report=xml:coverage-$CIRRUS_TASK_NAME.xml --codecov
```

Comment:
What's the point of reporting coverage data to codecov if we don't do anything with it? I would just get rid of the coverage reporting altogether.

Reply:
We do, https://github.com/seantis/pytest-codecov is a plugin to upload the coverage data.

Reply:
I know, but where is the data visualized?

Reply:
https://app.codecov.io/gh/mesonbuild/meson-python/ and the badge in the README and docs. I'd like to improve the coverage, so IMO it's definitely useful.

Reply:
There must be something wrong with the codevcov.io: it does not point to the source code in this repository. and looking at the details for any given file just shows "There was a problem getting the source code from your provider. Unable to show line by line coverage." 

I don't expect that running the code on different flavors of Linux will exercise different code paths, and this doubles the CI execution time. I prefer fast feedback loops. Are we sure we really need this? Does codecov provide a measure of how much each "measurement" from each test run contributes to the total coverage?